go_internal/main.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"sync"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+type DBMedia struct {
+	gorm.Model
+	Id            string
+	Title         string
+	OriginalTitle string
+	Overview      string
+	ReleaseDate   string
+	// 	Genres        []Genre TODO: Should just be names of the genres
+	PosterPath string
+	Popularity int
+	//	Flatrate      Results // TODO: look at the python version
+	//	Free          Results // TODO: look at the python version
```

Comment:
We need to decide if the way we store these should change... It think the structure is a list of country/provider pairs(the country repeats for every provider bound to it), which isn't how TMDB stores it.

It would be nice if we made the switch to `dict[country, list[provider]]`, instead of list[dict[country, provider]] - That way we only store the country once per group of providers(or twice if we still split free and flatrate)

If we decide to change it, we will need to rewrite some Python logic

Reply:
I say lets bring out the sledgehammer ;D

Reply:
We need a follow-up issue/PR with the fixes to the Python code, or do that in Go as well ;D (but please don't do it in this PR it's already huge)

It just needs to update the MeiliSearch indexes

Reply:
Most of the work is done in #217 and #222